packages/fxa-auth-server/scripts/refund-unverified-accounts.ts

```diff
@@ -0,0 +1,445 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { Firestore, QueryDocumentSnapshot } from '@google-cloud/firestore';
+import program from 'commander';
+import { StripeHelper } from '../lib/payments/stripe';
+import Stripe from 'stripe';
+import Container from 'typedi';
+
+import { setupProcessingTaskObjects } from '../lib/payments/processing-tasks-setup';
+import { AppConfig, AuthFirestore } from '../lib/types';
+import { promises as fs } from 'fs';
+
+const pckg = require('../package.json');
+
+type SubscriptionData = {
+  uid: string;
+  subscription: Stripe.Subscription;
+};
+
+type OutputData = {
+  uid: string;
+  email: string;
+  accountCreated: number;
+  customerId: string;
+  subscriptionId: string;
+  subscriptionCreated: number;
+  subscriptionStatus: string;
+  subscriptionPlan: string;
+  subscriptionInvoiceId: string;
+  refunded: boolean;
+};
+
+type ErrorOutput = {
+  uid: string;
+  email: string;
+  errorMessage: string;
+  error?: string;
+};
+
+class MergedAccountError extends Error {
+  public uid: string;
+  constructor(message: string, uid: string) {
+    super(message);
+    this.message = message;
+    this.uid = uid;
+  }
+}
+
+const isOfStatus = (
+  keyInput: string
+): keyInput is Stripe.Subscription.Status => {
+  return [
+    'active',
+    'canceled',
+    'incomplete',
+    'incomplete_expired',
+    'past_due',
+    'trialing',
+    'unpaid',
+  ].includes(keyInput);
+};
+
+const parseDryRun = (dryRun: boolean | string) => {
+  return `${dryRun}`.toLowerCase() !== 'false';
+};
+
+const parseDateForFirestore = (date: number) => Math.floor(date / 1000);
+
+const parseStartDate = (date: string): number =>
+  parseDateForFirestore(
+    date ? Date.parse(date) : Date.parse('2022-12-23T00:00:00')
+  );
+
+const parseEndDate = (date: string): number =>
+  parseDateForFirestore(
+    date ? Date.parse(date) : Date.parse('2022-12-31T23:59:59')
+  );
+
+const parseSubscriptionStatus = (
+  status: string
+): Stripe.Subscription['status'] | undefined => {
+  if (status) {
+    if (!isOfStatus(status)) {
+      throw new Error(
+        `Please provide a valid status. Invalid status: ${status}`
+      );
+    } else {
+      return status as Stripe.Subscription['status'];
+    }
+  } else {
+    return undefined;
+  }
+};
+
+function getCollectionNames() {
+  const config = Container.get(AppConfig);
+  const prefix = `${config.authFirestore.prefix}stripe-`;
+  return {
+    customerCollection: `${prefix}customers`,
+    subscriptionCollection: `${prefix}subscriptions`,
+    invoiceCollection: `${prefix}invoices`,
+  };
+}
+
+const getUidFromPath = (doc: QueryDocumentSnapshot) => {
```

Comment:
Why are some of these declared as functions, and some declared with arrow functions?

Reply:
No real reason. I've seen in places, for some one off functions that immediately return a value, an arrow function reads a little easier. (Nonsensical-ish example below)

```
function temp(isTrue) {
  return isTrue ? true : false;
}

vs

const temp = (isTrue) => isTrue ? true : false;
```